### ISSUE TITLE
Complete WebGL line rendering

### ISSUE THREAD

**Issue by ianthomas23** [bokeh/bokeh#11050]
Following the switch to ReGL, improvements are needed to complete WebGL line rendering.  Here they are grouped together for convenience, to be ticked off as they are completed.

- [ ] Antialiasing on thin lines is not quite correct.  Best seen on lines that are nearly horizontal or vertical.
- [ ] Dashes with butt end caps have slightly dodgy antialiasing at joins.
- [ ] Dashes with round or square end caps need rethink at joins.
- [ ] Spurious end caps on mobile devices.  May be due to lower floating-point precision in shaders.
- [ ] Overlapping lines have artifacts, probably alpha-related. Look into discarding pixels instead of alpha=0?
- [ ] Line dash pattern can have odd number of entries; repeat (concatenate) if so.

**Comment by ianthomas23** [comment]
Simpler reproducer for the line regression using 5 rather than 100k points:
```python
from bokeh.plotting import figure, show
from bokeh.layouts import row

dx = 1e-3
x = [0, 1, 1, 1+dx, 2]
y = [0, 0, 1, 0.8,  0.8]

ps = []
for backend in ('canvas', 'webgl'):
    p = figure(width=300, height=300, title=f"{backend} dx={dx}",
               output_backend=backend)
    p.line(x, y, color="#22aa22", line_width=12)
    ps.append(p)

show(row(ps))
```
![Screenshot_2021-03-26 Bokeh Plot](https://user-images.githubusercontent.com/580326/112633388-35acf080-8e31-11eb-96c6-288abc308c76.png)

**Comment by bryevdv** [comment]
@ianthomas23 checking in on this issue too, would it be good to move to `next` ?

**Comment by ianthomas23** [comment]
@bryevdv Yes, bump this to `next` milestone.  I should be back up and up and running in early September.

**Comment by bryevdv** [comment]
Thanks @ianthomas23 !
